aten/src/ATen/native/sparse/SparseFactories.cpp

```diff
@@ -91,5 +95,48 @@ Tensor spdiags(
   return result_coo;
 }
 
+Tensor spdiags_backward(
+    const Tensor& grad_out,
+    const Tensor& offsets,
+    IntArrayRef input_shape) {
+  auto offsets_1d = offsets.dim() == 0 ? offsets.unsqueeze(0) : offsets;
+
+  auto n_diag = input_shape.size() == 2 ? input_shape[0] : 1;
+  auto n_col_in = input_shape.size() == 2 ? input_shape[1] : input_shape[0];
+  AT_ASSERT(grad_out.dim() == 2);
+  AT_ASSERT(offsets_1d.size(0) == n_diag);
+  auto grad_in_options = grad_out.options().layout(Layout::Strided);
+  // zeros since we are only going to se the non-zero elements
+  Tensor grad_in = at::zeros({n_diag, n_col_in}, grad_in_options);
+  auto grad_out_coo =
+      grad_out.layout() == Layout::Sparse ? grad_out : grad_out.to_sparse();
```

Comment:
I am assuming grad_out will have the same sparse layout at the output of the forward pass, (if called `spdiags(..., layout=torch.sparse_csr)` for example). So the conversion to coo is needed.

I am not sure if `grad_out` may be uncoalesced if the layout is coo. The kernel function as implemented here would not be correct for that case, but it would be a simple modification to make it correct for that case, as we are already zero initializing `grad_in`.